src/mapping/query/function/split.rs

```diff
@@ -1,8 +1,86 @@
 use super::prelude::*;
 use bytes::Bytes;
-use lazy_static::lazy_static;
 use std::collections::BTreeMap;
-use std::ops::Deref;
+use std::convert::TryFrom;
+
+pub struct DynamicRegex {
+    pattern: String,
+    multiline: bool,
+    insensitive: bool,
+    global: bool,
+    compiled: Option<Result<regex::Regex>>,
+}
+
+impl DynamicRegex {
+    pub fn new(pattern: String, multiline: bool, insensitive: bool, global: bool) -> Self {
+        Self {
+            pattern,
+            multiline,
+            insensitive,
+            global,
+            compiled: None,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn is_global(&self) -> bool {
+        self.global
+    }
+
+    pub fn compile(&mut self) -> Result<&regex::Regex> {
+        if self.compiled.is_none() {
+            self.compiled = Some(
+                regex::RegexBuilder::new(&self.pattern)
+                    .case_insensitive(self.insensitive)
+                    .multi_line(self.multiline)
+                    .build()
+                    .map_err(|err| format!("invalid regex {}", err)),
+            );
+        }
+
+        self.compiled
+            .as_ref()
+            // We know this unwrap is safe because we have just populated the Option above.
+            .unwrap()
```

Comment:
You should be able to use `get_or_insert` here, to achieve the same without having to unwrap.

Reply:
Oh, nice!